lib/ansible/modules/cloud/azure/azure_rm_batchaccount_info.py

```diff
@@ -0,0 +1,251 @@
+        self.resource_group = None
+        self.name = None
+        self.tags = None
+        super(AzureRMBatchAccountInfo, self).__init__(self.module_arg_spec, supports_tags=False)
+
+    def exec_module(self, **kwargs):
+        for key in self.module_arg_spec:
+            setattr(self, key, kwargs[key])
+        self.mgmt_client = self.get_mgmt_svc_client(BatchManagementClient,
+                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+
+        if self.resource_group is not None and self.name is not None:
+            self.results['items'] = self.get()
+        elif self.resource_group is not None:
+            self.results['items'] = self.list_by_resource_group()
+        else:
+            self.results['items'] = self.list_by_subscription()
+        return self.results
+
+    def get(self):
+        response = None
+        results = []
+        try:
+            response = self.mgmt_client.batch_account.get(resource_group_name=self.resource_group,
+                                                          account_name=self.name)
+            self.log("Response : {0}".format(response))
+        except CloudError as e:
+            self.log('Could not get info for Batch Account.')
+
+        if response and self.has_tags(response.tags, self.tags):
+            results.append(self.format_response(response))
+
+        return results
+
+    def list_by_resource_group(self):
+        response = None
+        results = []
+        try:
+            response = self.mgmt_client.batch_account.list_by_resource_group(resource_group_name=self.resource_group)
+            self.log("Response : {0}".format(response))
+        except CloudError as e:
+            self.log('Could not get info for Batch Account.')
+
+        if response is not None:
+            for item in response:
+                if self.has_tags(item.tags, self.tags):
+                    results.append(self.format_response(item))
+
+        return results
+
+    def list_by_subscription(self):
+        response = None
+        results = []
+        try:
+            response = self.mgmt_client.batch_account.list()
+            self.log("Response : {0}".format(response))
+        except CloudError as e:
+            self.log('Could not get info for Batch Account.')
+
+        if response is not None:
+            for item in response:
+                if self.has_tags(item.tags, self.tags):
+                    results.append(self.format_response(item))
+
+        return results
+
+    def format_response(self, item):
+        d = item.as_dict()
+        d = {
+            'id': d['id'],
+            'resource_group': self.resource_group,
+            'name': d['name'],
+            'location': d['location'],
+            'account_endpoint': d['account_endpoint'],
+            'auto_storage_account': d['auto_storage']['storage_account_id'],
+            'key_vault': d['key_vault_reference']['id'],
```

Comment:
It seems that the response doesn't contain `key_vault_reference` field.

Reply:
Fixed using `get()`.